system_metrics_collector/test/system_metrics_collector/test_linux_cpu_measurement.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <memory>
+#include <mutex>
+
+#include "../../src/system_metrics_collector/linux_cpu_measurement_node.hpp"
+
+namespace
+{
+constexpr const char proc_sample_1[] =
+  "cpu  22451232 118653 7348045 934943300 5378119 0 419114 0 0 0\n";
+constexpr const char proc_sample_2[] =
+  "cpu  22451360 118653 7348080 934949227 5378120 0 419117 0 0 0\n";
+constexpr const std::chrono::milliseconds TEST_PERIOD =
+  std::chrono::milliseconds(50);
+constexpr const double CPU_ACTIVE_PERCENTAGE = 2.8002699055330633;
+}
+
+class TestLinuxCpuMeasurementNode : public LinuxCpuMeasurementNode
+{
+public:
+  TestLinuxCpuMeasurementNode(
+    const std::string & name,
+    const std::chrono::milliseconds measurement_period,
+    const std::string & publishing_topic)
+  : LinuxCpuMeasurementNode(name, measurement_period, publishing_topic,
+      PeriodicMeasurementNode::DEFAULT_PUBLISH_WINDOW)
+  {}
+  virtual ~TestLinuxCpuMeasurementNode() = default;
+
+  double periodicMeasurement() override
+  {
+    LinuxCpuMeasurementNode::periodicMeasurement();
+  }
+
+private:
+  ProcCpuData makeSingleMeasurement() override
+  {
+    ProcCpuData toReturn;
+    if (first) {
+      toReturn = LinuxCpuMeasurementNode::processLine(proc_sample_1);
+    } else {
+      toReturn = LinuxCpuMeasurementNode::processLine(proc_sample_2);
+    }
+
+    first = !first;
+    return toReturn;
+  }
+
+  bool first{true};
+};
+
+class LinuxCpuMeasurementTestFixture : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    const char * const argv = "d";
+    rclcpp::init(1, &argv);
```

Comment:
```suggestion
    rclcpp::init(0, nullptr);
```

This code is wrong (the same issue exists in other tests). If `rclcpp` keeps a copy of the pointer `argv`, it will segv if accessed after exiting `SetUp` scope. As this does not pass anything in particular, let's keep it safe and just set this to null.